main-app/app/scripts/game/services/game-state.js

```diff
@@ -0,0 +1,37 @@
+(function () {
+    'use strict';
+    angular.module('Tombola.Games.Bingo90.Game').
```

Comment:
Oho. Try to think of this as an object, and it  simplifies itself:
1. A ticket has three lines. The house prize has been won if all lines have 0 squares to go. The to-go value is the lowest line TOGO value until the line prize is won. After the prize is won the togo value us the sum of each line's TOGOs 
2. A line has 9 squares - the TOGO value is the sum of all the TOGOs for each square within it
3. A square has a value (null if none, otherwise the number). Give it a 'called' value - set to true if the number is called. The TOGO value is true if the value is not null and not called.

Then give the ticket a "make call" function  which takes the call number and fills in the ticket. Also not the first line called is not *necessary* a line prize - in real game other players can get there first. Use the server responses for line and house prizes.